File: nexus_os/apps/web/scripts/verify_routes.ts

import axios from 'axios';

const routes = [
    '/',
    '/gotham',
    '/ontology',
    '/erp',
    '/crm',
    '/hcm',
    '/protocols',
    '/builder',
    '/aip/terminals',
    '/warp',
    '/ciso/sec-001',
    '/supply-chain',
    '/finance'
];

const baseUrl = 'http://localhost:3000';

async function checkRoutes() {
    console.log('Checking routes...');
    let success = true;
    for (const route of routes) {
        try {
            const url = `${baseUrl}${route}`;
            const start = Date.now();
            const res = await axios.get(url);
            const duration = Date.now() - start;
            console.log(`[PASS] ${route} - Status: ${res.status} - Time: ${duration}ms`);
        } catch (error: any) {
            success = false;
            if (error.response) {
                console.error(`[FAIL] ${route} - Status: ${error.response.status}`);
            } else {
                console.error(`[FAIL] ${route} - Error: ${error.message}`);
            }
        }
    }
    if (!success) {
        process.exit(1);
    }
}

checkRoutes();

File: nexus_os/apps/web/verify_routes.py

import urllib.request
import urllib.error

base_url = "http://localhost:3000"
routes = [
    "/",
    "/login",
    "/workspace",
    "/gotham",
    "/foundry",
    "/ontology",
    "/warp",
    "/apollo",
    "/erp",
    "/crm",
    "/hcm",
    "/itsm",
    "/supply-chain",
    "/ciso",
    "/autobidder",
    "/garage",
    "/builder",
    "/protocols",
    "/command-center",
    "/aip"
]

print(f"{'ROUTE':<20} | {'STATUS':<10} | {'MESSAGE'}")
print("-" * 50)

for route in routes:
    url = f"{base_url}{route}"
    try:
        with urllib.request.urlopen(url) as response:
            print(f"{route:<20} | {response.getcode():<10} | OK")
    except urllib.error.HTTPError as e:
        print(f"{route:<20} | {e.code:<10} | {e.reason}")
    except urllib.error.URLError as e:
        print(f"{route:<20} | {'ERR':<10} | {e.reason}")
    except Exception as e:
        print(f"{route:<20} | {'ERR':<10} | {e}")
